hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -492,23 +493,26 @@ private void callDisconnectionOperation(ClientEndpointImpl endpoint) {
                 // "a disconnected client is reconnected back to same node"
                 return;
             }
-            ClientDisconnectionOperation op = createClientDisconnectionOperation(clientUuid, memberUuid);
-            operationService.run(op);
+            int partitionId = nodeEngine.getPartitionService().getPartitionId(clientUuid);
+            ClientDisconnectionOperation op = createClientDisconnectionOperation(partitionId, clientUuid, memberUuid);
+            operationService.execute(op);
 
             for (Member member : memberList) {
                 if (!member.localMember()) {
-                    op = createClientDisconnectionOperation(clientUuid, memberUuid);
+                    op = createClientDisconnectionOperation(partitionId, clientUuid, memberUuid);
                     operationService.send(op, member.getAddress());
                 }
             }
         }
     }
 
-    private ClientDisconnectionOperation createClientDisconnectionOperation(String clientUuid, String memberUuid) {
+    private ClientDisconnectionOperation createClientDisconnectionOperation(int partitionId,
+                                                                            String clientUuid, String memberUuid) {
         ClientDisconnectionOperation op = new ClientDisconnectionOperation(clientUuid, memberUuid);
         op.setNodeEngine(nodeEngine)
                 .setServiceName(SERVICE_NAME)
                 .setService(this)
+                .setPartitionId(partitionId)
```

Comment:
I guess we should also call `setValidateTarget(false)` here. Another thing is, what if the target partition is migrating?

Reply:
Ok. we handle `validateTarget` on the operation class.